src/data_type/mod.rs

```diff
@@ -2285,6 +2285,34 @@ impl DataType {
             (Err(_), Err(_)) => Err(Error::other("No common variant")),
         }
     }
+
+    pub fn lower_bound(&self) -> Option<DataType> {
```

Comment:
We have the convention of naming the same way the generic method and specific one:
Why not call this `min` ?